microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import PurePath, PurePosixPath
+from dataclasses import dataclass
+from lisa.executable import Tool
+from lisa.messages import TestStatus
+from typing import Any
+from lisa.tools.remote_copy import RemoteCopy
+from lisa.tools.tar import Tar
+import os
+from lisa.util import MissingPackagesException
+
+@dataclass
+class KselftestResult:
+    version: str = ""
+    name: str = ""
+    status: TestStatus = TestStatus.QUEUED
+    exit_value: int = 0
+
+class Kselftest(Tool):
+    @property
+    def command(self) -> str:
+        return str(self._command)
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        # command to run kselftests
+        self._command = self.get_tool_path(use_global=True) / "kselftest-packages/run_kselftest.sh > kselftest-results.txt"
```

Comment:
` > kselftest-results.txt` can be moved to run all function for clarity and keep only uptil `.sh`

Reply:
If one line is 100bytes, it looks like around 470 lines? It's ok to put in log. You can save the output to a file at the same time, if it needs to be parsed.

Reply:
> If one line is 100bytes, it looks like around 470 lines? It's ok to put in log. You can save the output to a file at the same time, if it needs to be parsed.

@squirrelsc the logs are around 47,000 lines.